cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -45,80 +109,302 @@ def evaluate_dynamic_defaults(self, raw_data, **kwargs):
+
+
+class EfsSchema(BaseSchema):
+    """Represent the schema of EFS."""
+
+    encrypted = fields.Bool()
+    kms_key_id = fields.Str()
+    performance_mode = fields.Str(validate=validate.OneOf(["generalPurpose", "maxIO"]))
+    throughput_mode = fields.Str(validate=validate.OneOf(["provisioned", "bursting"]))
+    provisioned_throughput = fields.Int(validate=validate.Range(min=0, max=1024))
+    file_system_id = fields.Str(validate=validate.Regexp(r"^fs-[0-9a-z]{8}$|^fs-[0-9a-z]{17}$"))
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return EfsConfig(**data)
+
+    @validates_schema
+    def validate_existence_of_mode_throughput(self, data, **kwargs):
+        """Validate the conditional existence requirement between throughput_mode and provisioned_throughput."""
+        throughput_mode = data.get("throughput_mode")
+        provisioned_throughput = data.get("provisioned_throughput")
+        if throughput_mode != "provisioned" and provisioned_throughput:
+            raise ValidationError(
+                message="When specifying 'provisioned_throughput', the 'throughput_mode' must be set to 'provisioned'",
+                field_name="ThroughputMode",
+            )
+
+        if throughput_mode == "provisioned" and not provisioned_throughput:
+            raise ValidationError(
+                message="When specifying 'throughput_mode' to 'provisioned',"
+                " the 'provisioned_throughput' option must be specified",
+                field_name="ProvisionedThroughput",
+            )
+
+
+class FsxSchema(BaseSchema):
+    """Represent the schema of FSX."""
+
+    storage_capacity = fields.Int()
+    deployment_type = fields.Str(validate=validate.OneOf(["SCRATCH_1", "SCRATCH_2", "PERSISTENT_1"]))
+    imported_file_chunk_size = fields.Int()
+    export_path = fields.Str()
+    import_path = fields.Str()
+    weekly_maintenance_start_time = fields.Str(validate=validate.Regexp(r"^[1-7]:([01]\d|2[0-3]):([0-5]\d)$"))
+    automatic_backup_retention_days = fields.Int(validate=validate.Range(min=0, max=35))
+    copy_tags_to_backups = fields.Bool()
+    daily_automatic_backup_start_time = fields.Str(validate=validate.Regexp(r"^([01]\d|2[0-3]):([0-5]\d)$"))
+    per_unit_storage_throughput = fields.Int(validate=validate.OneOf(FSX_SSD_THROUGHPUT + FSX_HDD_THROUGHPUT))
+    backup_id = fields.Str(validate=validate.Regexp("^(backup-[0-9a-f]{8,})$"))
+    kms_key_id = fields.Str()
+    file_system_id = fields.Str(validate=validate.Regexp(r"^fs-[0-9a-z]{17}$"))
+    auto_import_policy = fields.Str(validate=validate.OneOf(["NEW", "NEW_CHANGED"]))
+    drive_cache_type = fields.Str(validate=validate.OneOf(["READ"]))
+    storage_type = fields.Str(validate=validate.OneOf(["HDD", "SSD"]))
+
+    @validates("imported_file_chunk_size")
+    def fsx_imported_file_chunk_size_validator(self, value):
+        """Validate ImportedFileChunkSize."""
+        if not 1 <= value <= 512000:
```

Comment:
You can use: [validate.Range](https://marshmallow.readthedocs.io/en/stable/marshmallow.validate.html#marshmallow.validate.Range) by passing a custom error message.

Reply:
Good catch. Done